packages/cloud-cognitive/src/components/InlineEdit/_inline-edit.scss

```
@@ -162,8 +184,11 @@
       position: absolute;
       top: 0;
       right: 0;
-      width: calc(2 * var(--#{$block-class}--size));
+      display: flex;
+      width: calc(3 * var(--#{$block-class}--size));
       height: 100%;
+      justify-content: flex-end;
```

`flex-end` doesn't seem quite right here: it stacks the ellipsis over to the right, miles away from the truncated text. I think the ellipsis should be right up at the left edge. With just an ellipsis and one button this can be accomplished with `space-between`, but actually I wonder whether the ellipsis really belongs inside the controls div at all? It isn't, after all, a control, and however many controls there are they probably want to stack from the right while the ellipsis wants to be at the extreme left abutting to the truncated text. Maybe the ellipsis and the controls div should be siblings in a wrapping div that occupies the blank space next to the text.

```suggestion
      justify-content: space-between;
```

---

@dcwarwick had a bit of a rework here as suggested now have

__right-input-elements containing ellipsis and toolbar, and __toolbar containing the validation icon and buttons.